api/envoy/extensions/http/custom_response/redirect_policy/v3/redirect_policy.proto

```diff
@@ -26,28 +26,75 @@ option (xds.annotations.v3.file_status).work_in_progress = true;
 // upstream.
 // [#next-free-field: 7]
 message RedirectPolicy {
-  // [#comment: TODO(pradeepcrao): Add the ability to specify the full uri, or just host or
-  // path rewrite for the redirection in the same vein as
-  // config.route.v3.RedirectAction]
-  // The host that will serve the custom response.
-  //
-  // Example:
-  //
-  // .. code-block:: yaml
-  //
-  //    uri: https://www.mydomain.com
-  //
-  string host = 1 [(validate.rules).string = {min_len: 1}];
-
-  // The path for the custom response.
-  //
-  // Example:
-  //
-  //  .. code-block:: yaml
-  //
-  //    path: /path/to/503_response.txt
-  //
-  string path = 2 [(validate.rules).string = {min_len: 1}];
+  // [#next-free-field: 7]
+  message RedirectAction {
```

Comment:
Taking a step back, what are the main differences between this and [RedirectAction](https://github.com/envoyproxy/envoy/blob/0eaebeb11bdaba33fbd722c0839b074a5c1a1919/api/envoy/config/route/v3/route_components.proto#L1624)?
Keeping parity (and reusing code) may be the right thing to do here, and would like to understand why this shouldn't be done.

Reply:
Please see my response to Yan's comment regarding the same. Please let me know if that sounds reasonable to you.

Reply:
Took a stab at using `RedirectAction` from route_components.proto PTAL